pkg/sql/tablemetadatacache/table_metadata_batch_iterator.go

```diff
@@ -184,52 +197,81 @@ func (batchIter *tableMetadataBatchIterator) fetchNextBatch(
 		}
 	}
 
+	// Collect the span stats for the tables in the batch.
+	res, err := batchIter.spanStatsFetcher.SpanStats(ctx, &roachpb.SpanStatsRequest{
+		Spans:  spans,
+		NodeID: "0", // Fan out.
+	})
+
+	if err != nil {
+		return false, err
```

Comment:
Should we also return `len(batchIter.batchRows) > 0` instead of false here, or do we want to abort the run if there is an error from the SpanStats call? It may be worth continuing to try and update as many rows as possible, instead failing the whole job

Reply:
Yeah that sounds good to me. Done.